libnmstate/nm/wired.py

```diff
@@ -23,61 +23,93 @@
 ZEROED_MAC = '00:00:00:00:00:00'
 
 
-def create_setting(iface_state, base_con_profile):
-    mtu = iface_state.get('mtu')
-    mac = iface_state.get(Interface.MAC)
+class WiredSetting(object):
+    def __init__(self, state):
+        self.mtu = state.get(Interface.MTU)
+        self.mac = state.get(Interface.MAC)
+
+        ethernet = state.get('ethernet', {})
```

Comment:
You get bonus points from me if you would also add a commit to convert these strings to constants. :-)

Reply:
Yes, it crossed my mind.

Reply:
Will post it in a separate PR, makes sense to change it all over the project.
BTW, It also revealed a bug we have: in the schema and some tests we have the ethernet specific properties under the interface subtree (flat) and not under the `ethernet` subtree.